.github/workflows/write_read_roundtrip_test/action.yml

```diff
@@ -0,0 +1,77 @@
+name: Lance File Write Read Round Trip Verification
+on:
+  workflow_dispatch:
+
+jobs:
+  run:
+    runs-on: "ubuntu-24.04"
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Install Build Requirements
+        run: |
+          sudo apt update
+          sudo apt install -y protobuf-compiler
+
+      - name: Check AWS_ACCESS_KEY_ID
+        run: |
+          if [ -z "${{ secrets.LANCEDB_S3_READER_ACCESS_KEY }}" ]; then
+            echo "AWS_ACCESS_KEY_ID is not set"
+            exit 1
+          else
+            echo "AWS_ACCESS_KEY_ID is set"
+          fi
+
+      - name: Check AWS_SECRET_ACCESS_KEY
+        run: |
+          if [ -z "${{ secrets.LANCEDB_S3_READER_SECRET }}" ]; then
+            echo "AWS_SECRET_ACCESS_KEY is not set"
+            exit 1
+          else
+            echo "AWS_SECRET_ACCESS_KEY is set"
+          fi
+
+      - name: Set up AWS CLI
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-access-key-id: ${{ secrets.LANCEDB_S3_READER_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.LANCEDB_S3_READER_SECRET }}
+          aws-region: us-east-2
+
+      - name: List files in S3 bucket
+        run: |
+          BUCKET_NAME="lance-ci-bucket"
+          aws s3 ls s3://$BUCKET_NAME
+
+      - name: Download files from S3 bucket
+        working-directory: .github/workflows/file_read_write_round_trip_verification
+        run: |
+          BUCKET_NAME="lance-ci-bucket"
+          aws s3 cp s3://$BUCKET_NAME . --recursive
+
+      - name: List files Downloaded
+        working-directory: .github/workflows/file_read_write_round_trip_verification
+        run: |
+          ls -l
+
+      - name: Build Lance
+        working-directory: ./python
+        run: |
+          python -m venv venv
+          source venv/bin/activate
```

Comment:
Why do you need a `venv`?

Reply:
for `maturin develop`, it throws error if we don't have a `virtualenv or conda environment` here.
```
>           maturin develop --release
💥 maturin failed
  Caused by: Couldn't find a virtualenv or conda environment, but you need one to use this command. For maturin to find your virtualenv you need to either set VIRTUAL_ENV (through activate), set CONDA_PREFIX (through conda activate) or have a virtualenv called .venv in the current or any parent folder. See https://virtualenv.pypa.io/en/latest/index.html on how to use virtualenv or use `maturin build` and `pip install <path/to/wheel>` instead.
```